### PR TITLE
cmd/libsnap: fix compile error on more restrictive gcc

### PR DESCRIPTION
**Pull request by zyga**
This fixes return type discrepancy between unsetenv and GDestroyNotify

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

**Comment by bboozzoo** [comment]
This will need a cherry-pick to 2.32.7 too

**Comment by mvo5** [comment]
@bboozzoo I don't think we run arch tests in 2.32 yet - so if this is only an issue on arch I think its fine to skip it for 2.32.

File: cmd/libsnap-confine-private/snap-test.c [hunk 1]
```diff
@@ -188,10 +188,10 @@ static void test_sc_snap_name_validate(void)
 	g_assert_null(err);
 
 	// In case we switch to a regex, here's a test that could break things.
-	const char *good_bad_name = "u-94903713687486543234157734673284536758";
-	char varname[41] = { 0 };
-	for (int i = 3; i <= 40; i++) {
-		g_assert_nonnull(strncpy(varname, good_bad_name, i));
+	const char good_bad_name[] = "u-94903713687486543234157734673284536758";
+	char varname[sizeof good_bad_name] = { 0 };
```
Comment (mvo5) [inline cmd/libsnap-confine-private/snap-test.c]: :heart: for the sizeof here!

File: cmd/libsnap-confine-private/snap-test.c [hunk 2]
```diff
@@ -188,10 +188,10 @@ static void test_sc_snap_name_validate(void)
...
+	for (size_t i = 3; i <= sizeof varname - 1; i++) {
```
Comment (mvo5) [inline cmd/libsnap-confine-private/snap-test.c]: (nitpick^2) we could use "i < sizeof varname" here which I personally find slightly more common in C but not really important.

File: cmd/libsnap-confine-private/utils-test.c [hunk 3]
```diff
@@ -99,6 +99,18 @@ static void test_die_with_errno(void)
 	g_test_trap_assert_stderr("death message: Operation not permitted\n");
 }
 
+// A variant of rmdir that is compatible with GDestroyNotify
+static void my_rmdir(const char *path)
+{
+  rmdir(path);
```
Comment (mvo5) [inline cmd/libsnap-confine-private/utils-test.c]: meta-question: should we log/error here if this (and chdir and unsetenv) return != 0 ? Given that this is tests only a error here might be ok?

**Comment by mvo5** [comment]
@bboozzoo Sure, happy to cherry-pick if this helps on fedora.

**Pull request merged**